### PR TITLE
List metadata available categories and available datasets

### PR DESCRIPTION
**Pull request by gwarmstrong**
Adds two new endpoints, `/metadata/category/available` and `/avaialable/dataset` that list the available metadata categories and available datasets, respectively.

### CODE REVIEW

File: microsetta_public_api/api/datasets.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+def available():
+    resources = get_resources()
+    escape = {schema.metadata_kw}
+    dataset_key = 'datasets'
+
+    if dataset_key not in resources:
+        return jsonify([]), 200
+
+    datasets = list(filter(
+        lambda x: x not in escape,
```
Comment (dhakim87) [inline microsetta_public_api/api/datasets.py:15]: If there are any security/privilege/privacy concerns with exposing the data behind schema.metadata_kw, the policy for denoting that it is hidden should be centralized to also check what happens when the user passes in the escaped key(s).  If we're perfectly happy to give them this data when a user passes in schema.metadata_kw, and this is just a facade to make things easier to the user, then the current implementation is totally fine.

**Review by dhakim87 — APPROVED** [review]
Looks fine to me.  One question though- why is schema.metadata_kw excluded from the available resources?  Are we worried that users might try to request that data?  If so, excluding it from the list of available resources isn't sufficient to block users from explicitly requesting it.

**Comment by gwarmstrong** [comment]
The escaped keywords, namely schema.metadata_kw are located in the same place as the dataset keywords we are looking for, but contain different information. E.g., the structure is something like this:

```
├── __metadata__
├── fecal
│   ├── alpha
│   ├── beta
│   └── pcoa
└── oral
    ├── alpha
    ├── beta
    └── pcoa
```

And the result we want back is `['fecal', 'oral']`, instead of `['__metadata__', 'fecal', 'oral']`.

No concerns about trying to block users from accessing information. The results API at this point is only concerned with totally public data.

File: microsetta_public_api/api/datasets.py (hunk 1 above)
Comment (gwarmstrong) [reply microsetta_public_api/api/datasets.py:15]: Just a facade to make things easier for the API user.

File: microsetta_public_api/api/datasets.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+from microsetta_public_api.config import schema
+
+
+def available():
+    resources = get_resources()
+    escape = {schema.metadata_kw}
+    dataset_key = 'datasets'
+
+    if dataset_key not in resources:
+        return jsonify([]), 200
```
Comment (wasade) [inline microsetta_public_api/api/datasets.py:12]: Should this be a 404?

**Review by wasade — COMMENTED** [review]
One question, non-blocking

File: microsetta_public_api/api/datasets.py (hunk 2 above)
Comment (gwarmstrong) [reply microsetta_public_api/api/datasets.py:12]: I think 200 with empty list is saying "I know you are looking for the names of datasets, but I have none, so here's any empty list".
Whereas 404 would be saying "You are looking for the names of datasets, but I cannot find anything like that, so here's an error".

I think the 200 case is more reflective of what is happening here.

**Pull request merged**